mobilecoind-dev-faucet/src/slam/mod.rs

```diff
@@ -0,0 +1,535 @@
+            for tries in 0..params.retries {
+                if self.stop_requested.load(Ordering::SeqCst) {
+                    return;
+                }
+                match tx_submitter.submit_tx(
+                    prepared_utxo.index,
+                    &tx,
+                    (prepared_utxo.index + node_index_offset + tries) as usize,
+                    logger,
+                ) {
+                    Ok(block_height) => {
+                        self.block_height.fetch_max(block_height, Ordering::SeqCst);
+                        self.num_submitted_txs.fetch_add(1, Ordering::SeqCst);
+                        // Submit a receipt over prepared_utxo.utxo.sender
+                        // This helps the backgorund worker thread track if this
+                        // payment succeeds, and avoid immediately requeuing the utxo
+                        let receipt = Self::get_receipts(&tx, recipient);
+                        if prepared_utxo.utxo_record.sender.send(receipt).is_err() {
+                            log::error!(logger, "Could not send SubmitTxResponse to worker thread");
+                        }
+                        return;
+                    }
+                    Err(SubmitTxError::Fatal) => {
+                        log::warn!(
+                            logger,
+                            "Fatal error when submitting tx #{}, giving up on this tx",
+                            prepared_utxo.index
+                        );
+                        return;
+                    }
+                    Err(SubmitTxError::Rebuild) => {
+                        log::debug!(logger, "Rebuilding tx #{}", prepared_utxo.index);
+                        break;
+                    }
+                    Err(SubmitTxError::Retry) => {
+                        std::thread::sleep(params.retry_period);
+                    }
+                }
+            }
+            std::thread::sleep(params.retry_period);
+        }
+        log::warn!(
+            logger,
+            "Built and submitted tx #{} a total of {} times without success, giving up",
+            prepared_utxo.index,
+            params.retries
+        );
+    }
+
+    /// Get a mobilecoind-compatible SubmitTxResponse containing appropriate
+    /// receipts
+    ///
+    /// This is needed for the background worker thread to track whether any of
+    /// these payments land and avoid queueing and utxos used by slam
+    fn get_receipts(
+        tx: &mc_transaction_core::tx::Tx,
+        recipient: &PublicAddress,
+    ) -> api::SubmitTxResponse {
+        // Construct sender receipt.
+        let mut sender_tx_receipt = api::SenderTxReceipt::new();
+        sender_tx_receipt.set_key_image_list(tx.key_images().iter().map(Into::into).collect());
+        sender_tx_receipt.set_tombstone(tx.prefix.tombstone_block);
+
+        // Construct receiver receipts.
+        let receiver_tx_receipts: Vec<_> = tx
```

Comment:
We can collect into a `protobuf::RepeatedField`, and that can be inferred here, so let's omit the type here and the `.into()` call below
```suggestion
        let receiver_tx_receipts = tx
```